server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Session.scala

```diff
@@ -31,17 +34,34 @@ import org.apache.james.jmap.model.State.State
 case class IsPersonal(value: Boolean)
 case class IsReadOnly(value: Boolean)
 
-object Account {
-  def apply(name: Username,
-            isPersonal: IsPersonal,
-            isReadOnly: IsReadOnly,
-            accountCapabilities: Set[_ <: Capability]): Account = {
+object AccountId {
+  def from(username: Username): Either[IllegalArgumentException, AccountId] = {
+    val sha1String = Hashing.sha1()
+      .hashString(username.asString(), StandardCharsets.UTF_8)
+      .toString
+    val refinedId: Either[String, Id] = refineV(sha1String)
```

Comment:
Why do we need to use `refineV`? Not saying it's wrong, just trying to understand, I'm still not comfortable with this

Reply:
that is the only way to convert a plain string into a strong typed string. There's no other way even you like it or not :)